src/fsharp/pars.fsy

```diff
@@ -5328,8 +5328,13 @@ opName:
   | LPAREN error rparen  
      {  reportParseErrorAt (lhs parseState) (FSComp.SR.parsErrorParsingAsOperatorName()); ident(CompileOpName "****", rhs parseState 2) }
 
+  /* (*) is a special case, since we need to distinguish it from the comments, so we compensate the position of operator itself, since parens are getting captured here as well, we do it so things like nameof(*) are demangled properly to '*' instead of op_Multiply */
   | LPAREN_STAR_RPAREN
-     {  ident(CompileOpName "*", rhs parseState 1) }
+     {  let m = rhs parseState 1
+        let p1 = mkPos m.StartLine (m.StartColumn + 1)
+        let p2 = mkPos m.EndLine (m.EndColumn - 1)
+        let r = mkFileIndexRange m.FileIndex p1 p2
+        ident(CompileOpName "*", r) }
```

Comment:
@dsyme I hope it is a normal approach for it - just to create a new range for it from the existing one.

Reply:
The range is wrong now, since the parens should be part of the identifier and are now placed outside.

Reply:
An approach we could probably have here is to add an additional SynPat.Paren node. It would require a workaround on our side, but that would be a more correct thing to do than changing parsed mode ranges.

Reply:
Aren't we doing the same with other operators (i.e. LPAREN operatorName rparen)?
We have the ident(CompileOpName $2, rhs parseState 2), which does not include parens, does it?

Reply:
> The range is wrong now, since the parens should be part of the identifier and are now placed outside. We rely on these ranges to include the nodes completely, and in this particular case the parens are a part of the grammar.

Aren't we doing the same with other operators (i.e. LPAREN operatorName rparen)?
We have the ident(CompileOpName $2, rhs parseState 2), which does not include parens, does it?

Reply:
It's not being done for expressions, like `(+) 1 2` where SynExpr.Paren is added.
For patterns it's done like this at least in some cases, and it's one of the things on my list. I've been fixing some of these range and node nesting inconsistencies in trees in previous PRs, but there's still work to do for cases like this.

Reply:
I think the best approach could be to handle LPAREN_STAR_RPAREN and other operators separately for patterns and expressions in cases where the parens are currently ignored in the resulting tree, and to produce correct SynPat.Paren(SynPat.Named ...) and SynExpr.Paren(SynExpr.Ident ...) nodes in each case.

Maybe opName could, in addition toIdent, return something like range option to create a Paren node for.

Reply:
Sigh...right. Thanks for the comment, I didn't think about it tbh. Gonna look at it over the weekend.